torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -75,6 +75,10 @@ static Variable applySlice(const Variable& self, int64_t dim, PyObject* slice, b
     // TODO: implement negative step
     throw ValueError("negative step not yet supported");
   }
+  if (start >= stop) {
+    // TODO: currently we don't have support for 0-sized dims, return size 0 tensor for now
+    return self.type().tensor();
```

Comment:
Can you add a test case for this?

Reply:
Is this the best place to put this logic? If I'm understanding this correctly, the benefits of this won't extend over to ATen (do zero-slices exist in aten?)